### PR TITLE
feat: update macpaw/extended_mock_http_client

### PR DESCRIPTION
**Pull request by alekseytupichenkov**
BREAKING CHANGE: macpaw/extended_mock_http_client to 3 version

### CODE REVIEW

File: tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php [hunk 1]
```diff
@@ -5,52 +5,64 @@
...
     }
 
+    public function testInitFailedHttpClientType(): void
+    {
+        $this->expectException(TypeError::class);
+
+        new ExtendedHttpMockClientCollection(['string']);
+    }
+
     public function testSetHandlersSuccess(): void
```
Comment (IiiigorGG) [inline tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php]:
```suggestion
    public function testSetHttpClientsSuccess(): void
```

File: tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php [hunk 2]
```diff
@@ -5,52 +5,64 @@
...
-        $clientCollection->setHandlers([[]]);
+        $clientCollection->setHttpClients([
+            $this->createHttpClient('test2@test.test')
+        ]);
 
-        $this->assertCount(1, $clientCollection->getHandlers());
+        $this->assertCount(1, $clientCollection->getHttpClients());
     }
 
     public function testSetHandlersFailed(): void
```
Comment (IiiigorGG) [inline tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php]:
```suggestion
    public function testSetHttpClientsFailed(): void
```

**Review by IiiigorGG — APPROVED** [review]
Looks great 🔥

File: tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php (hunk 1 above)
Comment (alekseytupichenkov) [reply tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php]: done

File: tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php (hunk 2 above)
Comment (alekseytupichenkov) [reply tests/Unit/Collection/ExtendedMockHttpClientCollectionTest.php]: done

**Pull request merged**